src/data_provider/CMakeLists.txt

```diff
@@ -135,13 +135,15 @@ elseif(UNIX AND NOT APPLE)
   set_target_properties(sysinfo PROPERTIES
           LINK_FLAGS "-static-libgcc -static-libstdc++")
 
-  if(NOT CMAKE_SYSTEM_NAME STREQUAL "AIX")
-    string(APPEND CMAKE_SHARED_LINKER_FLAGS " -Wl,-rpath=$ORIGIN")
-  else()
+  if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
     string(REPLACE ";" ":" CXX_IMPLICIT_LINK_DIRECTORIES_STR "${CMAKE_CXX_IMPLICIT_LINK_DIRECTORIES}")
     string(REPLACE ";" ":" PLATFORM_REQUIRED_RUNTIME_PATH_STR "${CMAKE_PLATFORM_REQUIRED_RUNTIME_PATH}")
     target_link_libraries(sysinfo -Wl,-blibpath:${INSTALL_PREFIX}/lib:${CXX_IMPLICIT_LINK_DIRECTORIES_STR}:${PLATFORM_REQUIRED_RUNTIME_PATH_STR})
-  endif(NOT CMAKE_SYSTEM_NAME STREQUAL "AIX")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "HP-UX")
+    string(APPEND CMAKE_SHARED_LINKER_FLAGS " -Wl,+b,${INSTALL_PREFIX}/lib")
```

Comment:
Is it not possible to link the library with a relative path? We introduced this capability https://github.com/wazuh/wazuh/pull/7327 to make Wazuh portable once built.